data/transformation_hatespeech_detection.py

```diff
@@ -0,0 +1,27 @@
+from typing import Tuple
+from datasets.arrow_dataset import Dataset
+import pandas as pd
+from type import PreprocessConfig, TestDataset, TrainDataset
+from configs.constants import Const
+from sklearn.model_selection import train_test_split
+
+
+def transform_hatespeech_detection_dataset(
+    dataset: Dataset, config: PreprocessConfig
+) -> Tuple[TrainDataset, TestDataset]:
+
+    train_data, test_data = train_test_split(
+        pd.DataFrame(dataset["train"]), test_size=0.2
```

Comment:
This would be better to be configured from the config!

Reply:
as in the ``test_size``

Reply:
sure, but test_size refers to the number of items in the test set. I can add this to the function itself as parameter?

Reply:
In pytorch and huggingface config there is a ``val_size`` which is a ratio.
We could rename this to ``val_ratio`` and add ``test_ratio`` to all configs.

Reply:
that's a model config, this is a data transformation step. not sure how we can share those?

Reply:
also, this is not val ratio! this is test ratio, there's no split in this dataset.

Reply:
Isn't ``train_test_split( pd.DataFrame(dataset["train"]), test_size=0.2`` splitting the data into train and test according to the ``test_ratio``? O

Reply:
it does! but it's not a validation set - it's the test set. there's no test set by default.

Reply:
To be honest, it's up to you. If you'd like to leave it hardcoded in the middle of a transformation then go ahead with this!

Reply:
I'll add it as a function parameter!

Reply:
done!

Reply:
👍